src/hooks/useRegisteredHotkey.ts

```diff
@@ -51,11 +53,49 @@ const isEditableTarget = (target: EventTarget | null) => {
   if (editableTags.includes(tagName)) {
     return true
   }
-  return Boolean(element.getAttribute?.("contenteditable"))
+
+  // Check contenteditable attribute - only true/"" means editable, not "false"
+  const contentEditable = element.getAttribute?.("contenteditable")
+  if (contentEditable === "true" || contentEditable === "") {
+    return true
+  }
+
+  // Check for any parent elements that are editable
+  let current = element.parentElement
+  for (let depth = 0; depth < MAX_PARENT_DEPTH && current; depth++) {
+    const tagName = current.tagName
+    if (editableTags.includes(tagName)) {
+      return true
+    }
+
+    const contentEditable = current.getAttribute?.("contenteditable")
+    if (contentEditable === "true" || contentEditable === "") {
+      return true
+    }
+
+    current = current.parentElement
+  }
+
+  return false
+}
+
+const isInputFocused = () => {
+  if (typeof document === "undefined") return false
+  const activeElement = document.activeElement
+  if (!activeElement) return false
+
+  const tagName = activeElement.tagName
+  if (tagName === "INPUT" || tagName === "TEXTAREA" || tagName === "SELECT") {
```

Comment:
i wonder if we can detect the current mouse position on the page and make sure it's within the 3d viewer in addition to this